src/full/Agda/Compiler/MAlonzo/Compiler.hs

```diff
@@ -1290,8 +1292,10 @@ callGHC = do
 
   let ghcBin = optGhcBin opts
 
+  -- Make GHC use UTF-8 when writing to stdout and stderr.
+  liftIO $ setEnv "GHC_CHARENC" "UTF-8"
```

Comment:
Could this be moved to `callCompiler` to keep the simple API?
We could have `callCompiler'` for a more parametrized version of the API?

Reply:
No, `callCompiler` is not only for calling GHC.